.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,34 @@
+name: Docker image build
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  docker-build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        lambda:
+          - blazer
+          - google-cidr
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # tag=v3.1.0
+
+      - uses: dorny/paths-filter@4512585405083f25c027a35db413c2b3b9006d50 # tag=v2.11.1
+        id: changes
+        with:
+          filters: |
+            lambda:
+              - '${{ matrix.lambda }}/**'
+              - '.github/workflows/docker-build.yml'
+
+      - name: Build Docker image
+        if: steps.changes.outputs.lambda == 'true'
+        working-directory: ${{ matrix.lambda }}
+        run: make docker
```

Comment:
To which Makefile does this refer to?

Reply:
The `matrix` job setup means that at this point, the `working-directory: ${{ matrix.lambda }}` line will be executing the `Makefile` within the lambda's folder.

When a new lambda gets added, it'll just be a matter of adding its folder to the matrix like so:

```yaml
matrix:
    lambda:
        - blazer
        - google-cidr
        - a-new-and-highly-fancy-lambda
```